### PR TITLE
Blog app - Views

### PR DESCRIPTION
**Pull request by NeckerFree**
**Project requirements**

- Implement the design from the sneak peek wireframes.
  - Styling is nice to have but focus on making sure everything functions first.

Login page
login

All Users page wireframe 
users

Single User page wireframe 
user

All Posts by a given User page wireframe 
posts

Single Post page wireframe
post

- Use methods that you have created in the [Blog App - processing data in models](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md).
- Hint: In order to see some posts or comments, you might need to add some data to your database as described [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project.md#use-models-to-insert-data).

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1 +1,6 @@
-<h1>Here is a List of Users</h1>
\ No newline at end of file
+<div id="users">
+  <% @users.each do |user| %>
+    <%= render partial: "shared/single_user", locals: { user: user } %>
+  <% end %>
```
Comment (AnselemOdims) [inline app/views/users/index.html.erb:4]: - [ ] Clicking on a user should take you to the user's show page but that is not the case with your functionality. Please kindly ensure that when a user is clicked in the index page, it redirects to the show page.

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,12 @@
...
+<div id="userTemplate">
+   <%= render partial: "shared/single_user", locals: { user: @user } %>
+   <div id="bio">
+   <h3>Bio</h3>
+   <p><%= @user.bio %></p>
+   </div>
+<% @user.posts.each do |post| %>
+     <%= render partial: "shared/single_post", locals: { post: post } %>
+<% end %>
+<%= button_to "See all posts"%>
```
Comment (AnselemOdims) [inline app/views/users/show.html.erb]: - [ ] The `See all Post` button should open the user's post index page. At the moment, it is not doing that. Please kindly ensure it redirects to the appropriate page

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1 +1,12 @@
-<h1>Show User Info for a given user</h1>
\ No newline at end of file
+<div id="userTemplate">
+   <%= render partial: "shared/single_user", locals: { user: @user } %>
+   <div id="bio">
+   <h3>Bio</h3>
+   <p><%= @user.bio %></p>
+   </div>
+<% @user.posts.each do |post| %>
+     <%= render partial: "shared/single_post", locals: { post: post } %>
```
Comment (AnselemOdims) [inline app/views/users/show.html.erb]: - [ ] It should use `#recent_post` method to render the user's 3 most recent post. At the moment you are rendering all user's posts. Please kindly make sure the user can only see the 3 most recent posts.

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,8 @@
-<h1>Here is a list of posts for a given user</h1>
\ No newline at end of file
+<div id="postTemplate">
+   <%= render partial: "shared/single_user", locals: { user: @user } %>
+<% @user.posts.each do |post| %>
+     <%= render partial: "shared/single_post", locals: { post: post } %>   
+     <%= render partial: "shared/collection_comments", locals: { post: post, user: @user } %>    
+<% end %>
```
Comment (AnselemOdims) [inline app/views/posts/index.html.erb:6]: - [ ] It should use `post.recent_comments` method to render 5 most recent comments. At the moment you are rendering all the post's comments. Please kindly ensure you only render the 5 most recent comments.

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -1 +1,8 @@
-<h1>Here is a list of posts for a given user</h1>
\ No newline at end of file
+<div id="postTemplate">
+   <%= render partial: "shared/single_user", locals: { user: @user } %>
+<% @user.posts.each do |post| %>
+     <%= render partial: "shared/single_post", locals: { post: post } %>
```
Comment (AnselemOdims) [inline app/views/posts/index.html.erb]: - [ ] Clicking on one of the posts should take you to the post's show page. Please kindly implement this.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @NeckerFree , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by descholar-ceo — APPROVED** [review]
Hi @NeckerFree , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 You implemented the requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**